src/python/gudhi/off_utils.pyx

```diff
@@ -25,21 +26,95 @@ __license__ = "MIT"
+
+    line = _get_next_line(file_desc)
+    # First line should be "OFF" (3d case) or "nOFF" (dD case)
+    if line.lower().startswith("off"):
+        dim = 3
+    elif line.lower().startswith("4off"):
+        dim = 4
+    elif line.lower().startswith("noff"):
+        # "nOFF" case, next line is the dimension
+        # can also contain nb_vertices, nb_faces nb_edges (can also be on the next line)
+        line = _get_next_line(file_desc)
+        digits = [int(s) for s in line.split() if s.isdigit()]
+        dim = digits[0]
+        if len(digits) > 1:
+            nb_vertices = digits[1]
+            # nb_faces =  digits[2]
+            # nb_edges =  digits[3] # not used - can be ignored
+            # nb_cells =  digits[4]
+    else:
+        raise ValueError(f"Inconsistent OFF header, got '{line.rstrip()}', should be 'OFF', '4OFF' or 'nOFF'")
+        
+    # nb_vertices can be already set by "nOFF" case, when 'dim nb_vertices nb_faces nb_edges' on the same line
+    if nb_vertices < 0:
+        # Number of points is the first number ("OFF" case) or the second one ("nOFF" case) of the second line
+        line = _get_next_line(file_desc)
+        digits = [int(s) for s in line.split() if s.isdigit()]
+        nb_vertices = digits[0]
+        # nb_faces =  digits[1]
+        # nb_edges =  digits[2] # not used - can be ignored
+        # nb_cells =  digits[3]
+    # If comments between header and the first point
+    line = _get_next_line(file_desc)
+    # Here the first line without comment is read - let's go back to the beginning of this line
+    file_desc.seek(file_desc.tell() - len(line))
+    return dim, nb_vertices
+
 def read_points_from_off_file(off_file=''):
     """Read points from an `OFF file <fileformats.html#off-file-format>`_.
 
     :param off_file: An OFF file style name.
     :type off_file: string
 
     :returns:  The point set.
-    :rtype: List[List[float]]
+    :rtype: numpy.ndarray
+
+    .. warning::
+        This function is using `numpy.loadtxt <https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html>`_
+        with `comments='#'` as an argument. It is advised to use a numpy version &ge; 1.23.0 if the `off_file` contains
+        lines without data or with comments in between the points.
     """
-    if off_file:
-        if os.path.isfile(off_file):
-            return read_points_from_OFF_file(off_file.encode('utf-8'))
-        else:
-            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT),
-                                    off_file)
+    with open(off_file) as input_file:
+        dim, nb_points = _read_off_file_header(input_file)
+        print(dim, nb_points)
+        # usecols=list(range(dim)) stands here to avoid comments at the end of line
+        # or colors that can be added in RGB format after the points, the faces, ...
+        return np.loadtxt(input_file, dtype=np.floating, comments='#',
+                          usecols=list(range(dim)), max_rows=nb_points)
```

Comment:
Doesn't `usecols=range(dim)` work? The documentation says it accepts a _sequence_.